src/Neo/Extensions/NeoTokenExtensions.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// NeoTokenExtensions.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace Neo.Extensions
+{
+    public static class NeoTokenExtensions
+    {
+        private const byte Prefix_Account = 20;
+
+        public static IEnumerable<(UInt160 Address, BigInteger Balance)> GetAccounts(this NeoToken neoToken, DataCache snapshot)
+        {
+            if (snapshot is null)
+                throw new ArgumentNullException(nameof(snapshot));
+
+            var kb = new KeyBuilder(neoToken.Id, Prefix_Account);
+            var prefixKey = kb.ToArray();
+
+            foreach (var (key, value) in snapshot.Seek(prefixKey, SeekDirection.Forward))
+            {
+                var keyBytes = key.ToArray();
+
+                if (keyBytes.AsSpan().StartsWith(prefixKey))
+                {
+                    var addressHash = new UInt160(keyBytes.AsSpan(prefixKey.Length));
+                    yield return new(addressHash, value.GetInteroperable<AccountState>().Balance);
+                }
+                else
+                    yield break;
+            }
+        }
+
+        public static BigInteger GetAccountBalance(this NeoToken neoToken, DataCache snapshot, UInt160 address)
```

Comment:
ERC20 extensions, and use `BalanceOf`?

Reply:
ERC20？

Reply:
This is for offline mode. `BalanceOf` requires and uses the `neo.vm`. This does not.

Reply:
> ERC20？

NEP17 xD